sciris/sc_fileio.py

```diff
@@ -2217,61 +2249,44 @@ def _unpickler(string=None, filename=None, filestring=None, die=None, verbose=Fa
-                        loaded = False
-                        while not loaded:
-                            try:
-                                obj = _RobustUnpickler(io.BytesIO(string), remapping=remapping).load() # And if that fails, throw everything at it
-                                loaded = True
-                            except Exception as E3b:
-                                from . import sc_versioning as scv # Here to avoid circular import
-                                remapping = scu.mergedicts(remapping)
-                                known = scv.known_deprecations()
-                                errstr = str(E3b)
-                                if errstr in known and auto_remap: # pragma: no cover
-                                    remapping.update(known[errstr]['fix'])
-                                    warnmsg = f'Fixing known unpickling deprecation "{errstr}"'
-                                    warnings.warn(warnmsg, category=UserWarning, stacklevel=2)
-                                else:
-                                    raise E3b
-                    except Exception as E4:
-                        try:
-                            if verbose: print(f'Robust failed ({str(E4)}), trying ultrarobust unpickler...')
-                            obj = _UltraRobustUnpickler(io.BytesIO(string), unpicklingerrors=[E1, E2, E3, E4]).load() # And if that fails, really throw everything at it
-                        except Exception as E5: # pragma: no cover
-                            errormsg = f'''
-All available unpickling methods failed:
-    Standard: {E1}
-     Encoded: {E2}
-        Dill: {E3}
-      Robust: {E4}
- Ultrarobust: {E5}'''
-                            raise Exception(errormsg)
+    methods = {
+        'Pickle': pkl.loads,
+        'Pandas': lambda string, **kwargs: pd.read_pickle(io.BytesIO(string), **kwargs),
+        'Dill': dill.loads,
+        'Encoded Pickle': lambda string, **kwargs: pkl.loads(string, encoding='latin1', **kwargs),
+        'Robust': lambda string, **kwargs: _RobustUnpickler.loads(string, remapping, **kwargs),
+        'Robust (renaming)': lambda string, **kwargs: _RenamingUnpickler.loads(string, remapping, **kwargs),
+        'Ultrarobust': lambda string, **kwargs: _UltraRobustUnpickler.loads(string, remapping, **kwargs),
+    }
+
+    # Methods that allow for loading an object without any failed instances (if die=True)
+    if method == 'pickle':
+        unpicklers = ['Pickle','Pandas','Encoded Pickle']
+    elif method == 'dill':
+        unpicklers = ['Dill']
+    else:
+        unpicklers = ['Pickle','Pandas','Dill','Encoded Pickle']
 
-    if isinstance(obj, Failed):
+    # If permitted, return an object that encountered errors in the loading process and therefore may not be valid
+    # Such an object might require further fixes to be made by the user
+    if not die:
```

Comment:
If `pickle.loads` fails and `pd.read_pickle` succeeds, the resulting object will have been correctly loaded even though the first attempt to load the object failed. Arguably, the same is true if `encoding='latin1'` also results in successful loading of the object. In that case, it seems like it would be useful for `die=True` to still allow loading the object, since the resulting object is perfectly good even though the first attempt to load the object failed. Whereas if any of the robust unpicklers were used, the replacement of modules or the resulting failed classes might make the object unusable without fixes on the user's end, which is the primary use case for setting `die=True` (i.e., if the user sets `die=True` then they can trust they won't need any error handling for badly loaded content in their file)".